pathfilters/pathfilters.go

```diff
@@ -119,11 +121,19 @@ func SDKFilter(sdk string, allowed bool) pathutil.FilterFunc {
 				buildConfigurations = append(buildConfigurations, target.BuildConfigurationList.BuildConfigurations...)
 			}
 
-			for _, buildConfiguratioon := range buildConfigurations {
-				sdk, err := buildConfiguratioon.BuildSettings.String("SDKROOT")
+			for _, buildConfiguration := range buildConfigurations {
+				sdk, err := buildConfiguration.BuildSettings.String("SDKROOT")
 				if err == nil {
 					sdkMap[sdk] = true
 				}
+
+				supportedPlatformsValue, err := buildConfiguration.BuildSettings.String("SUPPORTED_PLATFORMS")
```

Comment:
I was playing with Xcode to check if it will automatically update the `SDKROOT` value when I modify the supported platforms. And it does not do that. So I created an macOS app and then modified the supported platforms value to support every platform. The build will fail on any additional platforms and only succeed on macOS because the code does not support it officially. 

Based on this I think it would be great if we only run the extra logic when the `SDKROOT` value is `auto`:
```
if sdk != "auto" {
    continue
}
supportedPlatformsValue, err := buildConfiguration.BuildSettings.String("SUPPORTED_PLATFORMS")
...
```

Reply:
Nice finding, updated the code in: https://github.com/bitrise-io/go-xcode/pull/198/commits/2d86e20eaad4c8c8dde3606ccc0a3d5192ca4694